app/phone/src/main/java/dev/jdtech/jellyfin/PlayerActivity.kt

```diff
@@ -382,4 +420,72 @@ class PlayerActivity : BasePlayerActivity() {
             binding.playerView.useController = true
         }
     }
+
+    private fun startSleepTimer(timerDuration: Long) {
+        sleepJob?.cancel() // Cancel any existing timer job
+        sleepJob = CoroutineScope(Dispatchers.Main).launch {
+            delay(timerDuration)
+            binding.playerView.player?.playWhenReady = !binding.playerView.player?.playWhenReady!!
+            binding.playerView.findViewById<FrameLayout>(R.id.player_controls).visibility = View.GONE
+            binding.playerView.findViewById<FrameLayout>(R.id.sleep_mode_view).visibility = View.VISIBLE
+            window.clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
+            isControlsLocked = true
+        }
+        Toast.makeText(applicationContext, formatTimerMessage(timerDuration), Toast.LENGTH_SHORT).show()
+    }
+
+    private fun showSleepTimerDurationDialog() {
+        val timerTexts =
+            listOf("1 minute", "5 minutes", "10 minutes", "30 minutes", "1 hour", "2 hours")
+        val durationValues = longArrayOf(60000L, 300000L, 600000L, 1800000L, 3600000L, 7200000L)
+        var selectedDuration = durationValues[2] // Default: 10 minutes in milliseconds
+
+        val builder = MaterialAlertDialogBuilder(this)
+        builder.setTitle("Select Sleep Timer Duration")
+            .setSingleChoiceItems(
+                timerTexts.toTypedArray(),
+                timerTexts.indexOf("${selectedDuration / 60000} minutes"),
+            ) { _, which ->
+                selectedDuration = durationValues[which]
+            }
+            .setNegativeButton("Cancel") { dialog, _ ->
+                dialog.dismiss()
+                wasDialogShown = false
+            }
+            .setPositiveButton("OK") { dialog, _ ->
+                // Start the sleep timer with the selected duration
+                startSleepTimer(selectedDuration)
+                isSleepModeEnabled = true
+                dialog.dismiss()
+            }
+            .setCancelable(true)
+        wasDialogShown = true
+        builder.create().show()
+    }
+
+    private fun disableSleepMode() {
+        binding.playerView.player?.playWhenReady = true
+        window.addFlags (WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
```

Comment:
Lint issue here, just remove the space after `addFlag`.

Reply:
Thank you!